### ISSUE TITLE
Add an option to enclose String Values in the CSV Writer

### ISSUE THREAD

**Issue by joshuataylor** [pola-rs/polars#4439]
#### Describe your feature request

Having string values enclosed is a handy feature, especially when importing to other systems which accept CSV, but get confused when you have something like this:
```csv
foo,bar,baz
yes,true,1
```

It would be awesome to have this escaped like this:
```csv
foo,bar,baz
yes,"true",1
```

The API option could look like this:
```rust
    CsvWriter::new(&mut file)
        .with_string_field_enclosed(b'"')
```

Snowflake has this option when exporting CSVs, which makes it much easier:

```
FIELD_OPTIONALLY_ENCLOSED_BY = 'character' | NONE
Character used to enclose strings. Value can be NONE, single quote character ('), or double quote character ("). To use the single quote character, use the octal or hex representation (0x27) or the double single-quoted escape ('').

When a field contains this character, escape it using the same character. For example, if the value is the double quote character and a field contains the string A "B" C, escape the double quotes as follows:

A ""B"" C

Default: NONE
```

I'm not sure if this should be a boolean or a value the user can set (which I think it should be), similar to:
```rust
        .with_delimiter(b',')
```

I'd be happy to submit this change, I've been using it locally as I needed this.

Thoughts?

**Comment by ritchie46** [comment]
Yes, seems like a good option to me. This should be set by a boolean and the escaping should be double quotes as we want to only produce csv files that are compliant with: https://datatracker.ietf.org/doc/html/rfc4180

**Comment by alexander-beedie** [comment]
Closed by #10422 - use `quote_style="non_numeric"` 👍
